src/fuzzfetch/extract.py

```diff
@@ -5,75 +5,67 @@
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 # pylint: disable=too-many-statements
 
-from __future__ import absolute_import, division, print_function, unicode_literals
-
 import logging
 import os
 import shutil
 import stat
-import subprocess
 import tarfile
 import tempfile
 import zipfile
+from pathlib import Path
+from subprocess import DEVNULL, call, check_call, check_output
+from typing import Union
 
 from .path import onerror
 
 LOG = logging.getLogger("fuzzfetch")
 
 
-HDIUTIL_PATH = None
-P7Z_PATH = None
+HDIUTIL_PATH = shutil.which("hdiutil")
+P7Z_PATH = shutil.which("7z")
 
 
-def _extract_file(zip_fp, info, path):
-    """Extract files while explicitly setting the proper permissions"""
-    zip_fp.extract(info.filename, path=path)
-    out_path = os.path.join(path, info.filename)
+def extract_zip(zip_fn: Union[str, Path], path: Union[str, Path] = ".") -> None:
+    """Download and extract a zip artifact
 
-    perm = info.external_attr >> 16
-    perm |= stat.S_IREAD  # make sure we're not accidentally setting this to 0
-    os.chmod(out_path, perm)
+    Arguments:
+        zip_fn: path to zip archive
+        path: where to extract zip contents
+    """
+    path = Path(path)
 
+    def _extract_file(zip_fp, info):
+        """Extract files while explicitly setting the proper permissions"""
+        zip_fp.extract(info.filename, path=path)
+        out_path = path / info.filename
 
-def extract_zip(zip_fn, path="."):
-    """
-    Download and extract a zip artifact
+        perm = info.external_attr >> 16
+        perm |= stat.S_IREAD  # make sure we're not accidentally setting this to 0
+        out_path.chmod(perm)
 
-    Arguments:
-        zip_fn
-        path
-    """
-    global P7Z_PATH  # pylint: disable=global-statement
-    if P7Z_PATH is None:
-        P7Z_PATH = shutil.which("7z") or ""
     if P7Z_PATH:
-        subprocess.check_output([P7Z_PATH, "x", "-bd", f"-o{path}", zip_fn])
+        check_output([P7Z_PATH, "x", "-bd", f"-o{path}", zip_fn])
     else:
         with zipfile.ZipFile(zip_fn) as zip_fp:
             for info in zip_fp.infolist():
-                _extract_file(zip_fp, info, path)
+                _extract_file(zip_fp, info)
 
 
-def extract_tar(tar_fn, mode="", path="."):
-    """
-    Extract builds with .tar.(*) extension
+def extract_tar(
+    tar_fn: Union[str, Path], mode: str = "", path: Union[str, Path] = "."
```

Comment:
Minor not but you could create an alias for `Union[str, Path]` since it's used so frequently.  It could help prevent breaking up these long lines.